### ISSUE TITLE
Property does not exist on never with empty interface and type guard

### ISSUE THREAD

**Issue by xealot** [Microsoft/TypeScript#14059]
**TypeScript Version:**  2.1.1

**Code**

```ts
export interface Record {
    kind: string;
    name: string;
}

export interface UsefulType extends Record {}

export function isUseful(c: Record): c is UsefulType {
    return c.kind === "one";
}

export function doThing(record: Record) {
    if (isUseful(record)) {
        return null;
    }

    console.log(record.name);
    return record;
}
```

**Expected behavior:**

No errors.

**Actual behavior:**

`record` is converted to `never` after the `if`.

This seems to be because of the empty interface declaration. While I know it's not really that useful to declare an empty interface, it's useful for us to implicitly document this particular library and it really shouldn't cause an error.

If you add a member of any kind to the interface, the error clears up.

[Playground Link](https://www.typescriptlang.org/play/#src=export%20interface%20Record%20%7B%0D%0A%20%20%20%20kind%3A%20string%3B%0D%0A%20%20%20%20name%3A%20string%3B%0D%0A%7D%0D%0A%0D%0Aexport%20interface%20UsefulType%20extends%20Record%20%7B%7D%0D%0A%0D%0Aexport%20function%20isUseful(c%3A%20Record)%3A%20c%20is%20UsefulType%20%7B%0D%0A%20%20%20%20return%20c.kind%20%3D%3D%3D%20%22one%22%3B%0D%0A%7D%0D%0A%0D%0Aexport%20function%20doThing(record%3A%20Record)%20%7B%0D%0A%20%20%20%20if%20(isUseful(record))%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20null%3B%0D%0A%20%20%20%20%7D%0D%0A%0D%0A%20%20%20%20console.log(record.name)%3B%0D%0A%20%20%20%20return%20record%3B%0D%0A%7D)

**Comment by mhegazy** [comment]
Types in TypeScript are structural. The narrowing with the user-defined type guard will remove anything that looks like `UsefulType`, which happens to be the same as `Record` since it does not add any new properties. Removing type `Record`, you are left with `never`.

**Comment by xealot** [comment]
👍

**Comment by pranavanmaru** [comment]
`function isBook(text: Book | Magazine): text is Book {
    return (<Book>text).author !== undefined;
}

let readingMaterial: Book | Magazine = util.GetAllBooks()[0];

if (isBook(readingMaterial)) {
    console.log(`The book's author is ${readingMaterial.author}`);
}

else {
    console.log(`The magazine's publisher is ${readingMaterial.publisher}`);
}`

> TS2339	Property 'publisher' does not exist on type 'never'.

in my case though, I have a type that I'm left with after removing one type. Ideally this should make typescript narrow it down to Magazin in the else block. However that isn't happening.
